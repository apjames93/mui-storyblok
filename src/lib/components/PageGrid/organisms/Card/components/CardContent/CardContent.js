import React, { Suspense, lazy } from 'react';
import PropTypes from 'prop-types';
import { CardContent as MuiCardContent } from '@material-ui/core';
import { validComponents } from 'lib/utils/customProps';
import Storyblok from 'lib/utils/Storyblok';

import { renderComponentsWithBridge } from 'lib/utils/customComponents';

/* istanbul ignore next */
const Video = lazy(() => import('lib/components/PageGrid/atoms/Video/Video'));
/* istanbul ignore next */
const Typography = lazy(() => import('lib/components/PageGrid/molecules/Typography/Typography'));
/* istanbul ignore next */
const List = lazy(() => import('lib/components/PageGrid/organisms/List/List'));
/* istanbul ignore next */
const ListDropdown = lazy(() => import('lib/components/PageGrid/organisms/ListDropdown/ListDropdown'));
/* istanbul ignore next */
const Accordion = lazy(() => import('lib/components/PageGrid/organisms/Accordion/Accordion'));

const components = {
  Video,
  Accordion,
  Typography,
  List,
  ListDropdown,
};

const CardContent = ({
  content,
  rootClass,
  dataBlokC,
  dataBlokUid,
  storyblokClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <MuiCardContent
      className={`${styles.root} ${storyblokClass}`}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
    >
      {content.map((component, key) => (
        <Suspense fallback={<></>}>
          {renderComponentsWithBridge(components, component, key)}
        </Suspense>
      ))}
    </MuiCardContent>
  );
};

export default CardContent;

CardContent.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * Content passed to render
   * components:
    MuiTypography,
    MuiAccordion,
    MuiList,
   */
  content(props, propName, componentName) {
    const comps = [
      'Typography',
      'Accordion',
      'List',
    ];
    return validComponents(props, propName, componentName, comps);
  },
};

CardContent.defaultProps = {
  rootClass: [],
  content: [],
};
