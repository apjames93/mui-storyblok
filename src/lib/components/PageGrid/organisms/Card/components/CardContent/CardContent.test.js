import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CardContent from './CardContent';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "CardContent", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "Typography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'Typography',
      content: [{
        _editable: '<!--#storyblok#{"name": "Span", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'Span',
        content: 'text',
      }],
    }],
  };
  const comp = shallow(<CardContent {...props} />);
  return { comp, props };
}

describe('<CardContent />', () => {
  it('renders CardContent', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<CardContent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
