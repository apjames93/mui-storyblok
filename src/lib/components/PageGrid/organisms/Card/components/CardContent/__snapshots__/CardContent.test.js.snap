// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContent /> snapshot 1`] = `
<div
  className="MuiCardContent-root makeStyles-root-3 undefined"
/>
`;
